pages/opam-packaging.html

```diff
@@ -1,83 +1,153 @@
-</code></pre>
+<p>First, go to the GitHub page of <a href="https://github.com/coq/opam-coq-archive">opam-coq-archive</a> and <a href="https://help.github.com/articles/fork-a-repo/">make a fork</a>.</p>
 
-<p>The MD5 checksum is mandatory, and can be obtained with:</p>
+<p>The repository is now available in your personal GitHub space.
+Clone it locally by typing the following, where <code>$YOU</code> is your GitHub user name.
+The correct URL can also be found by clicking on the <em>clone or download</em> button
+on the GitHub page of your fork.
+</p>
 
-<pre><code>curl -L https://github.com/user/foo/archive/1.0.0.tar.gz | md5sum
+<pre><code>git clone https://github.com/$YOU/opam-coq-archive.git
+cd opam-coq-archive
 </code></pre>
 
-<p>Second, a text file called <code>descr</code> containing a short
-description of the package. In our case the contents of
-<code>released/packages/coq-foo/coq-foo.1.0.0/descr</code> are:</p>
+<p>Then create a directory named as follows</p>
 
-<pre><code>Foo is a Coq library doing wonders
+<pre><code>mkdir -p released/packages/coq-foo/coq-foo.1.0.0
 </code></pre>
 
-<p>Third, a text file called <code>opam</code> containing some metadata like build
-instructions and dependencies. In our case
-<code>released/packages/coq-foo/coq-foo.1.0.0/opam</code> contains:</p>
+<p>The Opam package is thus named <code>coq-foo</code> and the directory name
+is the package name followed by a dot followed by the version of the
+package.</p>
+
+<p>In that directory, create a text file called <code>opam</code>,
+which will contain the package's metadata (build instructions, dependencies, etc.).
+Here is a template for <code>released/packages/coq-foo/coq-foo.1.0.0/opam</code>:</p>
 
-<pre><code>opam-version: &quot;1.2&quot;
+<pre><code>opam-version: &quot;2.0&quot;
+synopsis: "A Coq library doing wonders" # One-line description
+description: """
+  Foo does bar with baz.
+""" # Longer description
+
+homepage: &quot;https://github.com/$YOU/foo&quot;
+dev-repo: &quot;git+https://github.com/$YOU/foo.git&quot;
+bug-reports: &quot;https://github.com/$YOU/foo/issues&quot;
 maintainer: &quot;your@email.address&quot;
-homepage: &quot;https://github.com/you/foo&quot;
-dev-repo: &quot;https://github.com/you/foo.git&quot;
-bug-reports: &quot;https://github.com/you/foo/issues&quot;
-authors: [&quot;Your Name&quot;]
+authors: [
+  &quot;You&quot;
+]
 license: &quot;MIT&quot;
+
+depends: [
+  &quot;coq&quot; {&gt;= &quot;8.5&quot; &amp; &lt; &quot;8.10~&quot;}
+]
+
 build: [
-  [&quot;coq_makefile -R . Foo -o Makefile *.v&quot;]
   [make &quot;-j%{jobs}%&quot;]
 ]
 install: [
   [make &quot;install&quot;]
 ]
-remove: [&quot;rm&quot; &quot;-R&quot; &quot;%{lib}%/coq/user-contrib/Foo&quot;]
-depends: [
-  &quot;coq&quot; {&gt;= &quot;8.5&quot; &amp; &lt; &quot;8.6~&quot;}
-]
+
+url {
+  src: "https://github.com/$YOU/foo/archive/1.0.0.tar.gz"
+  checksum: "sha256=133f2c6de1b9b35c0b26ab3e21fa86a351fffb415d20cae714b015f95fbc3725"
```

Comment:
Are there any downsides to using `sha256` instead of `md5`?

Reply:
I don't think so. md5 was kept in opam2 mostly for compatibility I think.